packages/ifpack/src/Ifpack_Hypre.cpp

```diff
@@ -299,7 +299,7 @@ int Ifpack_Hypre::ApplyInverse(const Epetra_MultiVector& X, Epetra_MultiVector&
   bool SameVectors = false;
   int NumVectors = X.NumVectors();
   if (NumVectors != Y.NumVectors()) IFPACK_CHK_ERR(-1);  // X and Y must have same number of vectors
-  if(X.Pointers() == Y.Pointers()){
+  if(X.Pointers() == Y.Pointers() || (NumVectors == 1 && X[0] == Y[0])){
```

Comment:
Is it possible to construct a noncontiguous column view of another MultiVector?  If so, what you actually want is to test whether each column pointer is the same.  Otherwise, I think this is OK.

Reply:
I thought that X[0] == Y[0] does that, checks for equality of column pointers. But I am not that familiar with Trilinos, the solution I made was actually proposed by Mr Heroux after I emailed him about the issue.